### PR TITLE
Add new key terms to chapters

### PR DESCRIPTION
**Pull request by Peter-Snipes**
Adding important terms to chapters 4, 21, 22, 27, 28, 30, 36, 45.

### CODE REVIEW

File: docs/operating-systems/schedule.md [hunk 1]
```diff
@@ -645,6 +654,7 @@ hide:
     - **Misdirected Write** - when data is written correctly to the disk storage but in the wrong memory address
     - **Cyclic Redundancy Check (CRC)** - remainder of binary division between a large binary number (D) and an agreed value of (k)
     - **Lost Write** - the device doesn't inform the upper layer that a write finished but the new changes never persisted into the next state, thus exposing the past state before the write
+    - **error correcting codes (ECC)** - used by the drive to determine whether the on-disk bits in a block are good or bad. in some cases the ecc will fix the bits but if it doesnt have enough information it will print error.
```
Comment (burgess01) [inline docs/operating-systems/schedule.md:657]: Please capitalize this so it matches the rest of the terms

**Review by burgess01 — CHANGES_REQUESTED** [review]
Please correct this one issue and then it will be ready to merge!

**Review by connellyw — APPROVED** [review]
Formating Resolved

**Review by burgess01 — APPROVED** [review]
This looks great, good job!

**Review by MAbraham24 — APPROVED** [review]
Looks good!

**Pull request merged**